src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -161,39 +192,93 @@ def iot_device_delete(client, hub_name, device_id, resource_group_name=None, eta
     return device_client.delete(device_id, etag)
 
 
-def iot_device_show_connection_string(client, hub_name, device_id=None, resource_group_name=None, top=20):
+def iot_device_show_connection_string(client, hub_name, device_id=None, resource_group_name=None,
+                                      key_type=KeyType.primary.value, top=20):
     resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
     if device_id is None:
         devices = iot_device_list(client, hub_name, resource_group_name, top)
         if devices is None:
             raise CLIError('No devices found in IoT Hub {}.'.format(hub_name))
         connection_strings = []
         for d in devices:
-            connection_string = _get_single_iot_device_connection_string(client, hub_name, d.device_id, resource_group_name)
-            connection_strings.append({"deviceId": d.device_id,
-                                       "connectionString": connection_string})
+            connection_string = _get_single_iot_device_connection_string(client, hub_name, d.device_id,
+                                                                         resource_group_name, key_type)
+            connection_strings.append({'deviceId': d.device_id,
+                                       'connectionString': connection_string})
         return connection_strings
     else:
-        connection_string = _get_single_iot_device_connection_string(client, hub_name, device_id, resource_group_name)
-        return {"connectionString": connection_string}
+        connection_string = _get_single_iot_device_connection_string(client, hub_name, device_id, resource_group_name,
+                                                                     key_type)
+        return {'connectionString': connection_string}
 
 
-def _get_single_iot_device_connection_string(client, hub_name, device_id, resource_group_name):
+def iot_device_send_message(client, hub_name, device_id, resource_group_name=None, data='Ping from Azure CLI',
+                            message_id=None, correlation_id=None, user_id=None):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+    device_client = _get_iot_device_client(client, resource_group_name, hub_name, device_id)
+    return device_client.send_message(device_id, data, message_id, correlation_id, user_id)
+
+
+def iot_device_receive_message(client, hub_name, device_id, resource_group_name=None, lock_timeout=60):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+    device_client = _get_iot_device_client(client, resource_group_name, hub_name, device_id)
+    result = device_client.receive_message(device_id, lock_timeout, raw=True)
+    if result is not None and result.response.status_code == 200:
```

Comment:
Because I need to retrieve properties from response headers.  Seems headers won't be returned without the raw flag.

---
In reply to: [88576596](https://github.com/Azure/azure-cli/pull/1344#discussion_r88576596) [](ancestors = 88576596)